packages/components/src/inline-edit/hook.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, negate, noop, flow } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+
+/**
+ * @param {import("react").ChangeEvent<HTMLInputElement>} event
+ */
+const cancelEvent = ( event ) => (
+	event.preventDefault(), event.stopPropagation(), event
+);
+
+/**
+ * @param {import("react").ChangeEvent<HTMLInputElement>} event
+ */
+const getEventValue = ( { target: { value } } ) => value;
+
+/**
+ * @param {Function[]} handlers
+ */
+const mergeEvent = ( ...handlers ) =>
+	/**
+	 * @param {import("react").ChangeEvent<HTMLInputElement>} event
+	 */
+	( event ) => handlers.forEach( ( handler = noop ) => handler( event ) );
+
+/**
+ * @typedef Props
+ * @property {(value: string) => boolean} validate predicate
+ * @property {Function} onWrongInput called when a validate predicate fails
+ * @property {Function} onCommit called on enter/blur
```

Comment:
These types can be improved slightly:

```suggestion
 * @property {( value: string ) => void} onWrongInput called when a validate predicate fails
 * @property {( value: string ) => void} onCommit called on enter/blur
```

Reply:
I think this would not be correct. `( value: string ) => void` would work only for a default value (which is noop), but we should be able to provide other kinds of signatures as well.

Reply:
What other signatures? These functions are used in this way in the hook itself so I would expect the signature to be predictable. Anyone consuming this API should know what type of function `onWrongInput` expects.

Reply:
Actually, other kinds of functions, different than the ones doing only side effects would not have any sense here so this signature is indeed preferred:)